dapps/v2Explorer/src/utils/UniversalProvider.tsx

```diff
@@ -0,0 +1,74 @@
+import '@walletconnect/react-native-compat';
+import UniversalProvider from '@walletconnect/universal-provider';
+
+// @ts-expect-error - `@env` is a virtualised module via Babel config.
+import {ENV_PROJECT_ID, ENV_RELAY_URL} from '@env';
+
+export let universalProvider;
+export let currentWCURI;
+export let universalProviderSession;
+
+export async function createUniversalProvider() {
+  console.log('[CONFIG] ENV_PROJECT_ID:', ENV_PROJECT_ID);
+  console.log('[CONFIG] ENV_RELAY_URL:', ENV_RELAY_URL);
+
+  try {
+    universalProvider = await UniversalProvider.init({
+      logger: 'info',
+      relayUrl: ENV_RELAY_URL,
+      projectId: ENV_PROJECT_ID,
+      metadata: {
+        name: 'React Native V2 dApp',
+        description: 'RN dApp by WalletConnect',
+        url: 'https://walletconnect.com/',
+        icons: ['https://avatars.githubusercontent.com/u/37784886'],
+      },
+    });
+
+    universalProvider.on('display_uri', uri => {
+      currentWCURI = uri;
+      console.log('UProvider URI:', uri);
+    });
+
+    // Subscribe to session ping
+    universalProvider.on('session_ping', ({id, topic}) => {
+      console.log(id, topic);
```

Comment:
Won't be clear without labelling in the log what event was actually triggered:

```suggestion
      console.log('session_ping', id, topic);
```